G1ANT.Addon.Azure/Extensions/TaskExtensions.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace G1ANT.Addon.Azure.Extensions
+{
+    public static class TaskExtensions
+    {
+        internal struct VoidTypeStruct { } //dummy struct - there is no non-generic version of TaskCompletionSource<TResult>
+        public static Task TimeoutAfter(this Task task, int millisecondsTimeout)
+        {
+            if (task.IsCompleted || (millisecondsTimeout == Timeout.Infinite))
+            {
+                return task;
+            }
+
+            TaskCompletionSource<VoidTypeStruct> tcs = new TaskCompletionSource<VoidTypeStruct>();
+
+            if (millisecondsTimeout == 0)
+            {
+                tcs.SetException(new TimeoutException());
+                return tcs.Task;
+            }
+
+            Timer timer = new Timer(state =>
+            {
+                var myTcs = (TaskCompletionSource<VoidTypeStruct>)state;
+                myTcs.TrySetException(new TimeoutException());
+            }, tcs, millisecondsTimeout, Timeout.Infinite);
+
+            task.ContinueWith((antecedent, state) =>
+            {
```

Comment:
Consider moving the code of this small anonymous method into separate named one. Just name it good.